hphp/runtime/vm/jit/vasm-ppc64.cpp

```diff
@@ -1022,13 +1013,27 @@ void lowerForPPC64(const VLS& e, Vout& v, vasm_src& inst) {             \
   else v << vasm_dst_imm{inst.s0, tmp2, inst.sf};                       \
 }
 
-X(cmpbim,  cmpl,  cmpli,  loadb)
+X(cmpbim,  cmpq,  cmpqi,  loadzbq)
 X(cmpqim,  cmpq,  cmpqi,  load)
-X(testbim, testq, testqi, loadb)
-X(testwim, testq, testqi, loadw)
-X(testlim, testq, testqi, loadl)
+X(testbim, testq, testqi, loadzbq)
 X(testqim, testq, testqi, load)
+#undef X
+
+#define X(vasm_src, vasm_dst_reg, vasm_dst_imm, vasm_load, vasm_exts)   \
+void lowerForPPC64(const VLS& e, Vout& v, vasm_src& inst) {             \
+  Vptr p = inst.s1;                                                     \
+  Vreg tmp; if (!e.allow_vreg()) tmp = Vreg(PhysReg(rAsm));             \
+  Vreg tmp2 = e.allow_vreg() ? v.makeReg() : Vreg(PhysReg(rAsm));       \
+  Vreg tmp3 = v.makeReg();                                              \
+  v << vasm_load{p, tmp2};                                              \
+  v << vasm_exts{tmp2, tmp3};                                           \
+  if (patchImm(inst.s0, v, tmp))                                        \
+    v << vasm_dst_reg{tmp, tmp3, inst.sf};                              \
```

Comment:
Braces are required for multiline conditionals.

Reply:
Done.